parsl/serialize/facade.py

```diff
@@ -0,0 +1,183 @@
+from parsl.serialize.concretes import *  # noqa: F403,F401
+from parsl.serialize.base import METHODS_MAP_DATA, METHODS_MAP_CODE
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ParslSerializer(object):
+    """ ParslSerializer gives you one interface to multiple serialization libraries
+    """
+    __singleton_instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """ Make ParslSerializer a singleton using the fact that there's only one instance
+        of a class variable
+        """
+        if ParslSerializer.__singleton_instance is None:
+            ParslSerializer.__singleton_instance = object.__new__(cls, *args, **kwargs)
+        return ParslSerializer.__singleton_instance
+
+    def __init__(self):
+        """ Instantiate the appropriate classes
+        """
+        headers = list(METHODS_MAP_CODE.keys()) + list(METHODS_MAP_DATA.keys())
+        self.header_size = len(headers[0])
+
+        self.methods_for_code = {}
+        self.methods_for_data = {}
+
+        for key in METHODS_MAP_CODE:
+            self.methods_for_code[key] = METHODS_MAP_CODE[key]()
+            self.methods_for_code[key].enable_caching(maxsize=128)
+
+        for key in METHODS_MAP_DATA:
+            self.methods_for_data[key] = METHODS_MAP_DATA[key]()
+
+    def _list_methods(self):
+        return self.methods_for_code, self.methods_for_data
+
+    def pack_apply_message(self, func, args, kwargs,
+                           buffer_threshold=None,
+                           item_threshold=None):
+        """Serialize and pack function and parameters
+
+        Parameters
+        ----------
+
+        func: Function
+            A function to ship
+
+        args: Tuple/list of objects
+            positional parameters as a list
+
+        kwargs: Dict
+            Dict containing named parameters
+
+        buffer_threshold: Ignored
```

Comment:
there's no need to have this and item_threshold if they're ignored? retaining ipyparallel.serialize invocation compatibilty doesn't seem like a goal?

Reply:
Sorry, this is a bad comment. I've added functionality to limit buffer size, which is checked in the serialize methods.
I'm removing `item_threshold`

Reply:
Sorry, this is a bad comment. I've added functionality to limit buffer size, which is checked in the serialize methods.
I'm removing `item_threshold`